### PR TITLE
[App Service] BREAKING CHANGE: `az functionapp`: Redact key value output on keys set

### PR DESCRIPTION
**Pull request by kamperiadis**
**Related command**
`az functionapp keys set`
`az functionapp function keys set`

**Description**
Currently, these two commands return the key value as part of the output. In this PR, we are redacting the key value to prevent accidental security leaks through CI logs. 

**Testing Guide**

1. Create functionapp `az functionapp create`
2. Add function app keys (`az functionapp keys set`) or function keys (`az functionapp function keys set`)
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -5381,15 +5381,21 @@ def update_host_key(cmd, resource_group_name, name, key_type, key_name, key_valu
...
+                                                                     key_name,
+                                                                     slot,
+                                                                     key=key_info)
+    else:
+        response = client.web_apps.create_or_update_host_secret(resource_group_name,
+                                                                name,
+                                                                key_type,
+                                                                key_name,
+                                                                key=key_info)
+    logger.warning('Keys have been redacted. Use keys list to view.')
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:5396]: It is suggested to specify the list command

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -5434,17 +5440,21 @@ def update_function_key(cmd, resource_group_name, name, function_name, key_name,
...
+                                                                         key_name,
+                                                                         slot,
+                                                                         key_info)
+    else:
+        response = client.web_apps.create_or_update_function_secret(resource_group_name,
+                                                                    name,
+                                                                    function_name,
+                                                                    key_name,
+                                                                    key_info)
+    logger.warning('Keys have been redacted. Use function keys list to view.')
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:5455]: It is suggested to specify the list command in the warning message.

**Comment by yanzhudd** [comment]
please rerun the test in live mode and upload the recording file.

**Comment by yanzhudd** [comment]
It is suggested to specify the command in the PR title:
[App Service] BREAKING CHANGE: `az functionapp`: Redact key value output on keys set

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (kamperiadis) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Done

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 2 above)
Comment (kamperiadis) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Done

**Comment by kamperiadis** [comment]
> please rerun the test in live mode and upload the recording file.

@yanzhudd I have addressed all your feedback. Can you please help review this PR again and merge if it looks good? Thank you!

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**